### PR TITLE
Bump ArgoCD default version to 5.37.0 (currently latest)

### PR DESCRIPTION
**Pull request by tsahiduek**
Argo CD helm version is outdated.
Helm chart version 5 introduced couple of changes, and I think we should upgrade it. It was tested on the pipeline-multi-env-gitops pattern

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by elamaran11 — APPROVED** [review]
LGTM

**Comment by shapirov103** [comment]
@tsahiduek the reason why I deferred upgrade to the latest version was because of the differences they introduced in handling private repositories. That is SSH key, github token, user/pwd authorization. Before we merge it, have you validated the current approach still works? I believe my initial research spotted some incompatibilities.

**Review by shapirov103 — CHANGES_REQUESTED** [review]
Please see my comment on private repo support. Happy to merge if it is not going to break the patterns for private repos.

**Comment by tsahiduek** [comment]
> 

Hi @shapirov103 ,
I've tested this with the GitOps Multi-env pattern that uses ArgoCD with SSH connection type --> https://github.com/aws-samples/cdk-eks-blueprints-patterns/blob/main/lib/pipeline-multi-env-gitops/index.ts#L250
Is this is what you meant?

**Review by shapirov103 — COMMENTED** [review]
@tsahiduek I also see indirect confirmation of the quality/compatibility here: https://github.com/aws-quickstart/cdk-eks-blueprints/issues/671#issuecomment-1623680147

LGTM

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**
